### PR TITLE
Make list-extensions work even if it's executed in a directory that doesn't have pom.xml

### LINKED ISSUE #8157: 1.3.0.Final quarkus:list-extensions doesn't work in non-quarkus project directory
1.3.0.Final quarkus:list-extensions doesn't work in non-quarkus project directory
`mvn io.quarkus:quarkus-maven-plugin:1.2.1.Final:list-extensions` works in any directory

Steps to reproduce:
```
mkdir foo && cd foo
mvn io.quarkus:quarkus-maven-plugin:1.3.0.Final:list-extensions
```

```
mvn io.quarkus:quarkus-maven-plugin:1.3.0.Final:list-extensions
[INFO] Scanning for projects...
...
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.3.0.Final:list-extensions 
  (default-cli) on project standalone-pom: Failed to list extensions: Neither project's 
  build file handler nor the project writer has been provided -> [Help 1]

```

### PR DESCRIPTION
**Pull request by geoand**
…oesn't have pom.xml

Fixes: #8157

### CODE REVIEW

File: independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensionsCommandHandler.java [hunk 1]
```diff
@@ -66,30 +66,41 @@ public QuarkusCommandOutcome execute(QuarkusCommandInvocation invocation) throws
...
+            final BuildFile buildFile = getBuildFileSafe(invocation);
             loadedExtensions.forEach(extension -> display(extension, installed, all, currentFormatter, buildFile));
 
-            if ("concise".equalsIgnoreCase(format)) {
+            if (buildFile != null) {
+                if ("concise".equalsIgnoreCase(format)) {
+                    if (buildFile instanceof GradleBuildFile) {
+                        System.out.println("\nTo get more information, append --format=full to your command line.");
+                    } else {
+                        System.out
```
Comment (rsvoboda) [inline independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensionsCommandHandler.java]: Can it be on 2 lines with nicer formatting?

File: independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensionsCommandHandler.java [hunk 2]
```diff
@@ -66,30 +66,41 @@ public QuarkusCommandOutcome execute(QuarkusCommandInvocation invocation) throws
...
         }
 
         return QuarkusCommandOutcome.success();
     }
 
+    private BuildFile getBuildFileSafe(QuarkusCommandInvocation invocation) {
+        try {
+            return invocation.getBuildFile();
+        } catch (Exception e) {
+            return null;
```
Comment (gastaldi) [inline independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensionsCommandHandler.java]: Maybe `log.debug` just in case?

**Review by rsvoboda — COMMENTED** [review]
Built and tried locally, `mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:list-extensions` works nicely in the directory where 1.3.0.Final based plugin fails.

Only small change of System.out.println and I will be happy ;)

File: independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensionsCommandHandler.java (hunk 2 above)
Comment (geoand) [reply independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensionsCommandHandler.java]: Done

File: independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensionsCommandHandler.java (hunk 1 above)
Comment (geoand) [reply independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensionsCommandHandler.java]: Like this?

Comment (rsvoboda) [reply independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensionsCommandHandler.java]: 👍

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**